### PR TITLE
GH-6723 - add AdaBoost documentation [nocheck]

### PR DESCRIPTION
**Pull request by valenad1**
GH-6723

### CODE REVIEW

**Review by hannah-tillman — DISMISSED** [review]
Did a few updates -- otherwise LGTM :)

**Comment by valenad1** [comment]
Hi @hannah-tillman, can you please approve once again? I added some section and the approve is dismissed.

**Review by hannah-tillman — DISMISSED** [review]
LGTM! Few minor changes and I split the `weak_learner` lists up so they're more accessibility-friendly. They look like this now:
Screen Shot 2023-10-06 at 10 13 18 AM

**Review by hannah-tillman — APPROVED** [review]

**Pull request merged**
